### ISSUE TITLE
Domain rules not working with subdomains and "host" match

### ISSUE THREAD

**Issue by IBBoard** [bitwarden/browser#957]
I've just started to use Bitwarden and I've found that it doesn't seem to apply domain rules to subdomains when using "host" match.

Credential URL: https://login.microfocus.com/nidp/idff/sso?id=suse&sid=1&option=credential
Login URL: https://build.opensuse.org/

(openSUSE uses the microfocus.com page for the login on its forums, but the build service uses its own domain with the same credentials).

I've tested some combinations and here's what I found:

Domain rule|Match|Result
---------------|-----|------------
microfocus.com, opensuse.org|Base domain|Login found
microfocus.com, opensuse.org|Host|Login not found
login.microfocus.com, build.opensuse.org|Host|Login not found

It seems that as soon as you use Host matching then it ignores the domain rules. It isn't even that it looks for perfect matches in the rules once you're on Host matching, because entering the subdomains (even though it says to enter the base domain) doesn't work either.

I could just use "base domain" matching, but I've got credentials on several subdomains and so host matching works better for me.

**Comment by kspearrin** [comment]
Currently, domain rules will only apply to base domain match types. This is because domain rules themselves are only evaluated on base domain values. Your case number 3 is not a valid domain rule. This is mentioned in the web vault when setting up domains rules:

> Enter a list of domains separated by commas. Only "base" domains are allowed. Do not enter subdomains. For example, enter "google.com" instead of "www.google.com". You can also enter "androidapp://package.name" to associate an android app with other website domains.

Host types can include subdomains and port numbers, which would never match a domain rule, so they are not evaluated.

**Comment by IBBoard** [comment]
I know the text says to enter base domains (hence why I said "even thought it says to enter the base domain") but I did it to check that the "Host" match didn't use those values differently.

The behaviour is non-intuitive for this situation, though. Nothing in the system makes it clear that when you swap from "Base domain" to "Host" as your global default strategy then the domain matches are completely ignored. It's not even obvious why the system needs to be base domains only and can't have subdomain matching (because some sites may want sharing on some subdomains but not others - like test.example.com and www.example.com using the same credentials but admin.example.com and five other subdomains don't, and the user doesn't want a long list of them and so uses Host matching)

I've realised that there are two workarounds for this.

1) Keep the default on Host but specifically set the credential URL to "Base domain" match
2) Add a second URL to the credential with the build.opensuse.org domain. I hadn't realised this was an option, as most managers just have a single URL (and it seems to make the custom domain matches a little redundant!)

It would be nice if Bitwarden could be more consistent (or clearer) in how it uses the domain matches, though.
